apps/metadata-editor/src/app/records/my-org-records/my-org-records.component.ts

```diff
@@ -16,29 +17,50 @@ import { AuthService } from '@geonetwork-ui/api/repository/gn4'
   imports: [CommonModule, TranslateModule, RecordsListComponent],
 })
 export class MyOrgRecordsComponent implements OnDestroy {
-  subscriptionAuthService: Subscription
-  subscriptionOrgService: Subscription
+  orgData: {
+    orgName: string
+    logoUrl: string
+    recordCount: number
+    userCount: number
+    userList: UserApiModel[]
+  }
+
+  public myOrgDataSubscription
 
   constructor(
+    private myOrgRecordsService: MyOrgService,
     public searchFacade: SearchFacade,
-    private authService: AuthService,
-    private orgService: OrganizationsServiceInterface
+    public orgService: OrganizationsServiceInterface,
+    public router: EditorRouterService
   ) {
     this.searchFacade.resetSearch()
-
-    this.subscriptionAuthService = this.authService.user$.subscribe((user) => {
-      this.searchByOrganisation({ name: user?.organisation })
-    })
+    this.myOrgDataSubscription = this.myOrgRecordsService.myOrgData$.subscribe(
+      (data) => {
+        this.orgData = data
+        this.searchByOrganisation({ name: data.orgName })
+      }
+    )
   }
 
   searchByOrganisation(organisation: Organization) {
-    this.subscriptionOrgService = this.orgService
+    this.orgService
       .getFiltersForOrgs([organisation])
       .subscribe((filters) => this.searchFacade.setFilters(filters))
   }
 
-  ngOnDestroy(): void {
-    this.subscriptionAuthService.unsubscribe()
-    this.subscriptionOrgService.unsubscribe()
+  getDatahubUrl(): string {
+    const url = new URL(
+      `${this.router.getDatahubSearchRoute()}`,
+      this.router.getDatahubSearchRoute().startsWith('http')
+        ? this.router.getDatahubSearchRoute()
+        : window.location.toString()
```

Comment:
Here you can simply use `window.location.toString()`. If a host is already present in the datahub URL it should be used.